crates/biome_module_graph/src/jsdoc_comment.rs

```diff
@@ -63,6 +63,26 @@ impl JsdocComment {
             && text.as_bytes().get(3).is_some_and(|c| *c != b'*')
             && text.ends_with("*/")
     }
+
+    /// Iterates all JSDoc comments preceeding the node.
+    pub fn for_each<F>(node: &JsSyntaxNode, mut func: F)
+    where
+        F: FnMut(&String),
```

Comment:
Change `&String` to `&str`. Using `&String` forces the caller to convert `str` to `String`, which allocates a string. However, you can extract a `str` from `String` at no cost.

Once you change it, it's possible that the compiler will fail, this means that you will have to change `filter_map.for_each` into a simple for loop

Reply:
Code is changed and branch rebased to reduce noise. 

I previously tried to follow the (mostly used) functional style but at this spot run into rust restrictions with borrowing. With the rewrite no `String` allocation happens at this spot.